core/src/apps/bitcoin/sign_tx/decred.py

```diff
@@ -160,7 +159,7 @@ def write_tx_output(
         script_pubkey: bytes,
     ) -> None:
         writers.write_uint64(w, txo.amount)
-        writers.write_uint16(w, txo.decred_script_version)
+        writers.write_uint16(w, DECRED_SCRIPT_VERSION)
```

Comment:
AFAICT this breaks spending from transactions that have `decred_script_version` other than 0, even if we are spending from a UTXO that has the correct version

Reply:
Uf, very good catch! I haven't realized these are called for TxOutputBinType as well even though it is typed correctly :facepalm:. 

So, is there a better way to write it than:

```
if isinstace(txo, TxOutputBinType):
  writers.write_uint16(w, txo.decred_script_version)
else:
  writers.write_uint16(w, DECRED_SCRIPT_VERSION)
```

?

Reply:
how about: `writers.write_uint16(w, getattr(txo, "decred_script_version", DECRED_SCRIPT_VERSION))`

Reply:
Looks alright as well, but I can't help myself that mine seems clearer.

Reply:
i don't have a problem with it either so go ahead